Sources/Core/GraphRequest.swift

```diff
@@ -0,0 +1,160 @@
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import UIKit
+
+// TODO: Resolve this
+typealias GraphPath = String
+// TODO: Resolve this
+typealias GraphRequestDataAttachment = Data
+
+struct GraphRequest {
+
+  /// The HTTPMethod to use for a graph request
+  enum HTTPMethod: String {
+    case get = "GET"
+    case post = "POST"
+    case delete = "DELETE"
+  }
+
+  struct Flags: OptionSet {
+    let rawValue: Int
+
+    static let none: Flags = Flags(rawValue: 1 << 0)
+
+    /// indicates this request should not use a client token as its token parameter
+    static let skipClientToken: Flags = Flags(rawValue: 1 << 1)
+
+    /// indicates this request should not close the session if its response is an oauth error
+    static let doNotInvalidateTokenOnError: Flags = Flags(rawValue: 1 << 2)
+
+    /// indicates this request should not perform error recovery
+    static let disableErrorRecovery: Flags = Flags(rawValue: 1 << 3)
+  }
+
+  /// The Graph API endpoint to use for the request, for example "me".
+  let graphPath: GraphPath
+
+  /// The request parameters.
+  let parameters: [String: Any]
+
+  /**
+   The HTTPMethod to use for the request, access rawValues for encoding purposes
+   ex: HTTPMethod.get.rawValue is "GET".
+  */
+  let httpMethod: HTTPMethod
+
+  /// An optional access token used by the request.
+  let accessToken: AccessToken?
+
+  /**
+   The Graph API version to use (e.g., "v2.0")
+   Defaults to `Settings.graphAPIVersion` if not specified during initialization
+  */
+  let version: String
+
+  var flags: Flags
+
+  /**
+   Initializes a new instance of a graph request.
+
+     - Parameter graphPath: the graph path (e.g., @"me")
+     - Parameter parameters: the optional parameters dictionary
+     - Parameter tokenString: an optional access token to use, must provide a token for paths that require a token
+     - Parameter version: the optional Graph API version (e.g., "v2.0"). nil defaults to `Settings.graphAPIVersion`.
+     - Parameter httpMethod: the HTTP method. Empty String defaults to `HTTPMethod.get`
+  */
+  init(
+    graphPath: GraphPath,
+    parameters: [String: Any] = [:],
+    accessToken: AccessToken? = AccessTokenWallet.shared.currentAccessToken,
+    version: String = Settings.graphAPIVersion,
+    httpMethod: GraphRequest.HTTPMethod = .get,
+    flags: GraphRequest.Flags = .none,
+    enableGraphRecovery: Bool = Settings.isGraphErrorRecoveryEnabled
+    ) {
+    self.graphPath = graphPath
+    self.parameters = parameters
+    self.accessToken = accessToken
+    self.version = version
+    self.httpMethod = httpMethod
+
+    var flags = flags
+    if !enableGraphRecovery {
+      flags = .disableErrorRecovery
+    }
+    self.flags = flags
+  }
+
+  var isGraphRecoveryDisabled: Bool {
+    return flags.contains(.disableErrorRecovery)
```

Comment:
Change `isGraphRecoveryDisabled` and `setGraphErrorRecoverability` into a single `isGraphErrorRecoveryEnabled` readwrite variable